src/app/qgisapp.cpp

```diff
@@ -1988,39 +2001,101 @@ QgsMessageBar *QgisApp::visibleMessageBar()
   }
 }
 
-QList<QgsVectorLayerRef> QgisApp::findBrokenWidgetDependencies( QgsVectorLayer *vl )
+const QList<QgsVectorLayerRef> QgisApp::findBrokenLayerDependencies( QgsVectorLayer *vl, QgsMapLayer::StyleCategories categories ) const
 {
   QList<QgsVectorLayerRef> brokenDependencies;
-  // Check for missing layer widget dependencies
-  for ( int i = 0; i < vl->fields().count(); i++ )
+
+  if ( categories.testFlag( QgsMapLayer::StyleCategory::Forms ) )
+  {
+    // Check for missing layer widget dependencies
+    for ( int i = 0; i < vl->fields().count(); i++ )
+    {
+      std::unique_ptr<QgsEditorWidgetWrapper> ww;
+      ww.reset( QgsGui::editorWidgetRegistry()->create( vl, i, nullptr, nullptr ) );
+      // ww should never be null in real life, but it is in QgisApp tests because
+      // QgsEditorWidgetRegistry widget factories is empty
+      if ( ww )
+      {
+        const QList<QgsVectorLayerRef> constDependencies { ww->layerDependencies() };
+        for ( const QgsVectorLayerRef &dependency : constDependencies )
+        {
+          const QgsVectorLayer *depVl { QgsVectorLayerRef( dependency ).resolveWeakly(
+                                          QgsProject::instance(),
+                                          QgsVectorLayerRef::MatchType::Name ) };
+          if ( ! depVl || ! depVl->isValid() )
+          {
+            brokenDependencies.append( dependency );
+          }
+        }
+      }
+    }
+  }
+
+
+  if ( categories.testFlag( QgsMapLayer::StyleCategory::Relations ) )
   {
-    std::unique_ptr<QgsEditorWidgetWrapper> ww;
-    ww.reset( QgsGui::editorWidgetRegistry()->create( vl, i, nullptr, nullptr ) );
-    // ww should never be null in real life, but it is in QgisApp tests because
-    // QgsEditorWidgetRegistry widget factories is empty
-    if ( ww )
+    // Check for layer weak relations
+    const QList<QgsWeakRelation> constWeakRelations { vl->weakRelations() };
+    for ( const QgsWeakRelation &rel : constWeakRelations )
     {
-      const auto constDependencies { ww->layerDependencies() };
-      for ( const QgsVectorLayerRef &dependency : constDependencies )
+      QgsRelation relation { rel.resolvedRelation( QgsProject::instance(), QgsVectorLayerRef::MatchType::Name ) };
+      QgsVectorLayerRef dependency;
+      bool found = false;
+      if ( ! relation.isValid() )
       {
-        const QgsVectorLayer *depVl { QgsVectorLayerRef( dependency ).resolveWeakly(
-                                        QgsProject::instance(),
-                                        QgsVectorLayerRef::MatchType::Name ) };
-        if ( ! depVl || ! depVl->isValid() )
+        // This is the big question: do we really
+        // want to automatically load the referencing layer(s) too?
```

Comment:
can we avoid the confusing referencing/referenced terminology and stick with the standard parent/child ? I know the api uses the confusing versions, but we can fix that easily for 4.0... comments will be harder to find and fix.

Reply:
For the record, the terminology was chosen on purpose. Parent/child is appropriate when we talk about compositions, but is misleading on aggregations and mostly on associations.